core/Context.h

```diff
@@ -67,6 +67,7 @@ class MutableContext final {
     MutableContext withOwner(SymbolRef sym) const;
     MutableContext withFile(FileRef file) const;
     ErrorBuilder beginError(LocOffsets loc, ErrorClass what) const;
+    ErrorBuilder beginIndexerError(LocOffsets loc, ErrorClass what) const;
```

Comment:
```suggestion
    // A version of `beginError` that's specific to the index phase of the pipeline, as it will record that index errors
    // have been seen on the file associated with the loc.
    ErrorBuilder beginIndexerError(LocOffsets loc, ErrorClass what) const;
```